plugins/woocommerce-blocks/tests/e2e/tests/product-collection/register-product-collection-tester/register-product-collection-tester.block_theme.spec.ts

```diff
@@ -0,0 +1,385 @@
+			await expect( products ).toHaveCount( 9 );
+
+			// Check if the preview button is visible
+			const previewButtonLocator = block.getByTestId(
+				SELECTORS.previewButtonTestID
+			);
+			await expect( previewButtonLocator ).toBeVisible();
+		} );
+	} );
+
+	test.describe( 'My Custom Collection with Advanced Preview', () => {
+		test( 'Clicking "My Custom Collection with Advanced Preview" should insert block and show 9 products', async ( {
+			pageObject,
+		} ) => {
+			await pageObject.createNewPostAndInsertBlock(
+				'myCustomCollectionWithAdvancedPreview'
+			);
+
+			await expect( pageObject.products ).toHaveCount( 9 );
+			await expect( pageObject.productImages ).toHaveCount( 9 );
+			await expect( pageObject.productTitles ).toHaveCount( 9 );
+			await expect( pageObject.productPrices ).toHaveCount( 9 );
+			await expect( pageObject.addToCartButtons ).toHaveCount( 9 );
+
+			await pageObject.publishAndGoToFrontend();
+			await expect( pageObject.products ).toHaveCount( 9 );
+		} );
+
+		test( 'Clicking "My Custom Collection with Advanced Preview" should show preview for 1 second', async ( {
+			pageObject,
+			editor,
+			page,
+		} ) => {
+			await pageObject.createNewPostAndInsertBlock(
+				'myCustomCollectionWithAdvancedPreview'
+			);
+			const previewButtonLocator = editor.canvas.getByTestId(
+				SELECTORS.previewButtonTestID
+			);
+
+			// The preview button should be visible
+			await expect( previewButtonLocator ).toBeVisible();
+
+			// Disabling eslint rule because we need to wait for the preview to disappear
+			// eslint-disable-next-line playwright/no-wait-for-timeout, no-restricted-syntax
+			await page.waitForTimeout( 1000 );
+
+			// The preview button should be hidden
+			await expect( previewButtonLocator ).toBeHidden();
+		} );
+
+		test( 'Should display properly in Product Catalog template', async ( {
+			pageObject,
+			editor,
+			page,
+		} ) => {
+			await pageObject.goToProductCatalogAndInsertCollection(
+				'myCustomCollectionWithAdvancedPreview'
+			);
+
+			const block = editor.canvas.getByLabel(
+				MY_REGISTERED_COLLECTIONS.myCustomCollectionWithAdvancedPreview
+					.label
+			);
+
+			// Check if the preview button is visible
+			const previewButtonLocator = block.getByTestId(
+				SELECTORS.previewButtonTestID
+			);
+			await expect( previewButtonLocator ).toBeVisible();
+
+			// Check if products are visible
+			const products = block
+				.getByLabel( BLOCK_LABELS.productImage )
+				.locator( 'visible=true' );
+			await expect( products ).toHaveCount( 9 );
+
+			// Disabling eslint rule because we need to wait for the preview to disappear
+			// eslint-disable-next-line playwright/no-wait-for-timeout, no-restricted-syntax
+			await page.waitForTimeout( 1000 );
+
+			// The preview button should be hidden after 1 second
+			await expect( previewButtonLocator ).toBeHidden();
+		} );
+	} );
+} );
+
+test.describe( 'Testing "usesReference" argument in "registerProductCollection"', () => {
```

Comment:
Let's use one parent `describe` per file for good practice. Also, I don't think we should reference implementation details (`usesReference`, `registerProductCollection`), as they can change, making this title confusing. What do you think if we structure this suite like ~this:

```js
test.describe("Product Collection registration", () => {
  test("registered collections are selectable", async () => {
    // ...
  });

  // ...

  [
    {
      id: 'myCustomCollectionWithProductContext',
      name: 'My Custom Collection - Product Context',
      previewLabelTemplate: [ 'woocommerce/woocommerce//single-product' ],
    },
    // ...
  ].forEach(collection => {
    test(`${collection.name}` preview is not displayed in posts`, async () => {
      // ...
      await pageObject.chooseCollectionInTemplate(collection.id)
      // ...
    });
  
    // ...
  });
});
```

The important thing is to keep the test structure as flat as possible. It makes the suite easier to read and understand.

cc: @kmanijak